frontend/src/pages/projects/screens/detail/notebooks/useNotebookImageData.ts

```diff
@@ -1,52 +1,94 @@
+import { NotebookImageData } from './types';
 
-const useNotebookImageData = (
-  notebook?: NotebookKind,
-): [
-  data: { imageStream: ImageStreamKind; imageVersion: ImageStreamSpecTagType } | null,
-  loaded: boolean,
-] => {
+const useNotebookImageData = (notebook?: NotebookKind): NotebookImageData => {
   const { dashboardNamespace } = useNamespaces();
-  const [images, loaded] = useImageStreams(dashboardNamespace);
+  const [images, loaded, loadError] = useImageStreams(dashboardNamespace, true);
 
   return React.useMemo(() => {
-    if (!notebook) {
-      return [null, false];
+    if (!notebook || !loaded) {
+      return [null, false, undefined];
     }
 
-    if (!loaded) {
-      return [null, false];
+    if (loadError) {
+      return [null, true, loadError];
     }
 
     const container: NotebookContainer | undefined = notebook.spec.template.spec.containers.find(
       (container) => container.name === notebook.metadata.name,
     );
     const imageTag = container?.image.split('/').at(-1)?.split(':');
 
+    // if image could not be parsed from the container, consider it deleted because the image tag is invalid
     if (!imageTag || imageTag.length < 2 || !container) {
-      return [null, true];
+      return [
+        {
+          imageAvailability: NotebookImageAvailability.DELETED,
+        },
+        true,
+        undefined,
+      ];
     }
 
     const [imageName, versionName] = imageTag;
     const imageStream = images.find((image) => image.metadata.name === imageName);
 
+    // if the image stream is not found, consider it deleted
     if (!imageStream) {
-      return [null, true];
+      // Get the image display name from the notebook metadata if we can't find the image stream. (this is a fallback and could still be undefined)
+      const imageDisplayName = notebook.metadata.annotations?.['opendatahub.io/image-display-name'];
+
+      return [
+        {
+          imageAvailability: NotebookImageAvailability.DELETED,
+          imageDisplayName,
+        },
+        true,
+        undefined,
+      ];
     }
 
     const versions = imageStream.spec.tags || [];
     const imageVersion = versions.find((version) => version.name === versionName);
 
+    // because the image stream was found, get its display name
+    const imageDisplayName = getImageStreamDisplayName(imageStream);
+
+    // if the image version is not found, consider the image stream deleted
     if (!imageVersion) {
-      return [null, true];
+      return [
+        {
+          loaded,
+          imageAvailability: NotebookImageAvailability.DELETED,
+          imageDisplayName,
+        },
+        true,
+        undefined,
+      ];
     }
 
-    return [{ imageStream, imageVersion }, loaded];
-  }, [images, loaded, notebook]);
+    // if the image stream exists and the image version exists, return the image data
+    return [
+      {
+        imageStream,
+        imageVersion,
+        loaded,
```

Comment:
Same here, remove it, I think you are using `loaded` outside of the `data` object.